src/IntoleranceSimulator.R

```diff
@@ -17,49 +17,60 @@ require(MASS)
 
 # hard-coded for quick prototyping
 
-protfile = 'structures/AF-P68133-F1-model_v2.pdb'
+protfile = '../structures/AF-P68133-F1-model_v2.pdb'
 
 pdb = read.pdb(protfile)
 
 # calculate residues pairwise distance matrix
-distmat = dm(pdb)
+#distmat = dm(pdb)
+
+# calculate distance between residues based on their carbon alpha
+distmat <- dm(pdb,inds="calpha")
 
 # distmat is upper triangular with empty diagonal: add diagonal, fill lower triangle, convert to matrix
 dismat = as.matrix(distmat)
 distmat[is.na(distmat)] = 0
 distmat = distmat + t(distmat)
 
-# spatial covariance function: use a Gaussian kernel for prototyping
+# calculate average distance between neighboring residues
+aadist = mean(diag(distmat[-nrow(distmat), -1]))
+
+# spatial covariance function
+# range:
+l = aadist * 5
 
-l=1
-sigma = exp(-0.5 * (distmat/l)^2)
+# use a Gaussian kernel for prototyping
+#sigma = exp(-0.5 * (distmat/l)^2)

```

Comment:
again, I'm assuming you're keeping this as an alternative code for sigma? It would be better to just keep the one you need

Reply:
Yes, ultimately the IntoleranceSimulator will have a nice interface to choose the spatial model and associated parameters.
For now, it is just fast prototyping

Reply:
sounds great